libraries/Microsoft.Bot.Builder.Core.Extensions/FileTranscriptStore.cs

```diff
@@ -0,0 +1,216 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Core.Extensions
+{
+    /// <summary>
+    /// The file transcript store stores transcripts in file system with each activity as a file.
+    /// </summary>
+    public class FileTranscriptStore : ITranscriptStore
+    {
+        private static JsonSerializerSettings jsonSettings = new JsonSerializerSettings()
+        {
+            Formatting = Formatting.Indented,
+            NullValueHandling = NullValueHandling.Ignore
+        };
+        protected string rootFolder;
+
+        public FileTranscriptStore(string folder)
+        {
+            this.rootFolder = folder ?? throw new ArgumentNullException("Missing folder");
+        }
+
+        /// <summary>
+        /// Log an activity to the transcript
+        /// </summary>
+        /// <param name="activity">activity to log</param>
+        /// <returns></returns>
+        public async Task LogActivity(IActivity activity)
```

Comment:
This isn't dong anything Async, and is probably generating a Build warning. Are there async versions of the file methods you could use?

Reply:
The interface is Async so wouldn't it have to still be marked as Async?  I agree about the need to use Async I/O here. To me that;s the real bug. 

---
In reply to: [181207228](https://github.com/Microsoft/botbuilder-dotnet/pull/410#discussion_r181207228) [](ancestors = 181207228)